modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/ManagementBar/RootOrganizationFilter.tsx

```diff
@@ -0,0 +1,199 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import {ClayDropDownWithItems} from '@clayui/drop-down';
+import {IItem} from '@clayui/drop-down/lib/drilldown/Menu';
+import ClayIcon from '@clayui/icon';
+import React, {
+	MouseEvent,
+	useCallback,
+	useContext,
+	useEffect,
+	useState,
+} from 'react';
+
+// @ts-ignore
+
+import ChartContext from '../ChartContext';
+
+const _toDropDownItems = (
+	items: TOrganization[],
+	onChange: (id: number) => void,
+	isRootOrgFromConfig = false
+): IItem[] =>
+	items.map(
+		({id: value, name: label}) =>
+			(({
+				checked: isRootOrgFromConfig,
+				disabled: isRootOrgFromConfig,
+				label,
+				onChange: () => onChange(Number(value)),
+				type: 'checkbox',
+				value: Number(value),
+			} as unknown) as IItem)
+	);
+
+const MAX_ROOT_ORGANIZATIONS = 10;
+
+const RootOrganizationFilter = () => {
+	const {
+		chartInstanceRef,
+		configRootOrganizationId,
+		fetchOrganizations,
+		organizations,
+		organizationsIds,
+		searchResult,
+		spritemap,
+	} = useContext(ChartContext);
+
+	const [changeApplied, setChangeApplied] = useState(true);
+	const [itemList, setItemList] = useState([] as IItem[]);
+	const [rootOrganizations, setRootOrganizations] = useState(
+		[] as TOrganization[]
+	);
+	const [searchValue, setSearchValue] = useState('');
+	const [selectedItems, setSelectedItems] = useState(
+		new Set() as Set<number>
+	);
+	const [rerenderKey, setRerenderKey] = useState(0);
+
+	const onItemsChange = useCallback((organizationId: number): void => {
+		setSelectedItems((current: Set<number>) => {
+			current[current.has(organizationId) ? 'delete' : 'add'](
+				organizationId
+			);
+
+			return new Set([...current]);
+		});
+
+		setChangeApplied(false);
+	}, []);
+
+	useEffect(() => {
+		setItemList((current: IItem[]) =>
+			current.map((item: IItem) => ({
+				...item,
+				checked: selectedItems.has(item.value),
+				disabled:
+					(selectedItems.size === MAX_ROOT_ORGANIZATIONS &&
+						!selectedItems.has(item.value)) ||
+					!!configRootOrganizationId,
+			}))
+		);
+	}, [configRootOrganizationId, selectedItems]);
+
+	useEffect(() => {
+		setItemList(
+			_toDropDownItems(
+				rootOrganizations,
+				onItemsChange,
+				!!configRootOrganizationId
+			)
+		);
+	}, [configRootOrganizationId, onItemsChange, rootOrganizations]);
```

Comment:
In case of changes on configRootOrganizationId I _think_ both useEffects are called and they performs the same setItemList but with different "data". Is it correct? Which one wins?

Reply:
They will trigger in order, but for the `configRootOrganizationId` dep this won't make any difference, as the drop-down item will be preselected and disabled (`_toDropDownItems`). Simply this won't reflect in the `selectedItems` state, but being not a real selection provided by the user, and in no way changeable, it also makes sense like this.

The other option would simply be to swap the declarative order of the 2 effects.

Reply:
(Changed and conditioned)